media/common/src/main/java/io/helidon/media/common/MediaContext.java

```diff
@@ -112,105 +140,202 @@ private Builder() {
+         *
+         * @param mediaSupport media support
+         * @param priority priority
+         * @return updated instance of the builder
+         */
+        public Builder addMediaSupport(MediaSupport mediaSupport, int priority) {
+            Objects.requireNonNull(mediaSupport);
+            services.addService((config) -> mediaSupport, priority);
             return this;
         }
 
         @Override
         public Builder addReader(MessageBodyReader<?> reader) {
-            readerContext.registerReader(reader);
+            builderReaders.add(reader);
             return this;
         }
 
         @Override
         public Builder addStreamReader(MessageBodyStreamReader<?> streamReader) {
-            readerContext.registerReader(streamReader);
+            builderStreamReaders.add(streamReader);
             return this;
         }
 
         @Override
         public Builder addWriter(MessageBodyWriter<?> writer) {
-            writerContext.registerWriter(writer);
+            builderWriters.add(writer);
             return this;
         }
 
         @Override
         public Builder addStreamWriter(MessageBodyStreamWriter<?> streamWriter) {
-            writerContext.registerWriter(streamWriter);
+            builderStreamWriter.add(streamWriter);
             return this;
         }
 
         /**
-         * Register a new stream reader.
-         * @param reader reader to register
+         * Whether defaults should be included.
+         *
+         * @param registerDefaults register defaults
          * @return this builder instance
          */
-        public Builder registerStreamReader(MessageBodyStreamReader<?> reader) {
-            readerContext.registerReader(reader);
+        public Builder registerDefaults(boolean registerDefaults) {
+            this.registerDefaults = registerDefaults;
             return this;
         }
 
         /**
-         * Register a new stream writer.
-         * @param writer writer to register
+         * Whether stack traces should be included in response.
+         *
+         * This is server side setting.
+         *
+         * @param includeStackTraces include stack traces
          * @return this builder instance
          */
-        public Builder registerStreamWriter(MessageBodyStreamWriter<?> writer) {
-            writerContext.registerWriter(writer);
+        public Builder includeStackTraces(boolean includeStackTraces) {
+            servicesConfig.computeIfAbsent(DEFAULTS_NAME, k -> new HashMap<>())
+                    .put(DEFAULTS_INCLUDE_STACK_TRACES, Boolean.toString(includeStackTraces));
             return this;
         }
 
         /**
-         * Whether defaults should be included.
+         * Whether system loader should be used to load {@link MediaSupportProvider} from classpath.
          *
-         * @param registerDefaults register defaults
+         * @param discoverServices use system loader
```

Comment:
Similar - it is not "system loader", it is "Java Service Loader"

Reply:
updated